tensorflow/core/kernels/dml_data_format_dim_map.cc

```diff
@@ -23,21 +23,58 @@ limitations under the License.
 
 namespace tensorflow {
 
+// Ensure that `src` and `dst` define a valid permutation.
+// Ops defined in this file assume that user specifies a permutation via two
+// string attributes. This check validates that these attributes properly define
+// it to prevent security vulnerabilities.
+static bool IsValidPermutation(const std::string& src, const std::string& dst) {
+  if (src.size() != dst.size()) {
+    return false;
+  }
+
+  std::map<char, bool> characters;
```

Comment:
Can this just be `std::array<bool, 256>` instead of allocating memory in the `map`? If you do so, be sure to cast the `char` to `uint8_t` before using as an index (some compilers are stupid and treat `char` as a signed value by default, which would be an out of bounds an access).

    for (const uint8_t c : src) {
      if (characters[c]) {
        return false;
      }
      characters[c] = true;
    }